### PR TITLE
add security and privacy chapter initial draft

### PR DESCRIPTION
**Pull request by eliasab16**
Before submitting your Pull Request, please ensure that you have carefully reviewed and completed all items on this checklist.

1. **Content**
    - [x] The chapter content is complete and covers the topic in detail.
    - [x] All technical terms are well-defined and explained.
    - [x] Any code snippets or algorithms are well-documented and tested.
    - [x] The chapter follows a logical flow and structure.
  
2. **References & Citations**
    - [x] All references are correctly listed at the end of the chapter.
    - [x] In-text citations are used appropriately and match the references.
    - [x] All figures, tables, and images have proper sources and are cited correctly.
  
3. **Quarto Website Rendering**
    - [x] The chapter has been locally built and tested using Quarto.
    - [x] All images, figures, and tables render properly without any glitches.
    - [ ] All images have a source or they are properly linked to external sites. 
    - [x] Any interactive elements or widgets work as intended.
    - [x] The chapter's formatting is consistent with the rest of the book.
  
4. **Grammar & Style**
    - [x] The chapter has been proofread for grammar and spelling errors.
    - [x] The writing style is consistent with the rest of the book.
    - [x] Any jargon is clearly explained or avoided where possible.
  
5. **Collaboration**
    - [ ] All group members have reviewed and approved the chapter.
    - [ ] Any feedback from previous reviews or discussions has been addressed.
  
6. **Miscellaneous**
    - [ ] All external links (if any) are working and lead to the intended destinations.
    - [ ] If datasets or external resources are used, they are properly credited and linked.
    - [ ] Any necessary permissions for reused content have been obtained.
  
7. **Final Steps**
    - [x] The chapter is pushed to the correct branch on the repository.
    - [ ] The Pull Request is made with a clear title and description.
    - [ ] The Pull Request includes any necessary labels or tags.
    - [ ] The Pull Request mentions any stakeholders or reviewers who should take a look.

### CODE REVIEW

File: privacy_security.qmd [hunk 1]
```diff
@@ -1,81 +1,1064 @@
...
+
+- **Knowledge:** while targeted attacks require a high degree offamiliarity with the data, subpopulation attacks require lessintimate knowledge in order to be effective.
+
+#### Case Study 1
+
+In 2017, researchers demonstrated a data poisoning attack against a popular toxicity classification model called Perspective [@hosseini2017deceiving]. This ML model is used to detect toxic comments online.
+
+The researchers added synthetically generated toxic comments with slight misspellings and grammatical errors to the model's training data. This slowly corrupted the model, causing it to misclassify increasing numbers of severely toxic inputs as non-toxic over time.
+
+After retraining on the poisoned data, the model's false negative rate increased from 1.4% to 27% - allowing extremely toxic comments to bypass detection. The researchers warned this stealthy data poisoning could enable the spread of hate speech, harassment, and abuse if deployed against real moderation systems.
```
Comment (arbass22) [inline privacy_security.qmd:199]: Wow this is super cool/effective/scary!

File: privacy_security.qmd [hunk 2]
```diff
@@ -1,81 +1,1064 @@
...
+
+- **Local Training:** Each hospital trains a machine learning model onits own patient data. This training happens locally, meaning thedata never leaves the hospital's servers.
+
+- **Model Sharing:** After training, each hospital only sends themodel (specifically, the parameters or weights of the model) to acentral server. They do not send any patient data.
+
+- **Aggregating Models:** The central server aggregates these modelsfrom all hospitals into a single, more robust model. This processtypically involves averaging the model parameters.
+
+- **Benefit:** The end result is a machine learning model that haslearned from a wide range of patient data without any of thatsensitive data having to be shared or leave its original location.
+
+#### Trade-offs
```
Comment (arbass22) [inline privacy_security.qmd:846]: Might be worth calling out that FL is still vulnerable to some types of attacks like data poisoning?

**Review by arbass22 — COMMENTED** [review]
Looks great overall just a few comments. I stopped putting the combined-word fixes once I realized it was a systemic problem. I wonder if someone's editor handled line breaks incorrectly but someone might need to go through this more thoroughly to fix these. Not a deal breaker though its still very clear, good job!

File: privacy_security.qmd [hunk 3]
```diff
@@ -1,81 +1,1064 @@
...
+
+To understand model inversion attacks, consider a facial recognition system used to grant access to secured facilities. The system is trained on a dataset of employee photos. An attacker, by observing the model's output to various inputs, could infer features of the original dataset. For example, if the model's confidence level for a particular face is significantly higher for a given set of features, an attacker might deduce that someone with those features is likely in the training dataset.
+
+The methodology of model inversion typically involves the following steps:
+
+- **Accessing Model Outputs:** The attacker queries the ML model withinput data and observes the outputs. This is often done through alegitimate interface, like a public API.
```
[the remark this answers is not in the document]
Comment (profvjreddi) [reply privacy_security.qmd:123]: Yea, this was me - oops. I did a big search and replace for some line breaks and messed it up and thought I had fixed it all. Thanks for catching these and noticing them. Will update.

File: privacy_security.qmd [hunk 4]
```diff
@@ -1,81 +1,1064 @@
...
+
+- **Training:** The ML model trains on this manipulated dataset anddevelops skewed understandings of the data patterns.
+
+- **Deployment:** Once the model is deployed, the corrupted trainingleads to flawed decision-making or predictable vulnerabilities theattacker can exploit.
+
+The impacts of data poisoning extend beyond just classification errors or accuracy drops. For instance, if incorrect or malicious data is introduced into a traffic sign recognition system's training set, the model may learn to misclassify stop signs as yield signs, which can have dangerous real-world consequences, especially in embedded autonomous systems like autonomous vehicles.
+
+Data poisoning can degrade the accuracy of a model, force it to make incorrect predictions or cause it to behave unpredictably. In critical applications like healthcare, such alterations can lead to significant trust and safety issues.
+
+There are four main categories of data poisoning [@oprea2022poisoning]:
```
[the remark this answers is not in the document]
Comment (profvjreddi) [reply privacy_security.qmd:179]: Thank you good sir! 🙏  Fixed.

File: privacy_security.qmd (hunk 1 above)
Comment (profvjreddi) [reply privacy_security.qmd:199]: Indeed, it is nuts! Some neat works out there.

**Review by mpstewart1 — CHANGES_REQUESTED** [review]

**Pull request merged**